src/main/java/tech/zerofiltre/blog/domain/course/use_cases/enrollment/Certificate.java

```diff
@@ -0,0 +1,108 @@
+            String courseTitle = courseProvider.getTitle(courseId);
+            String fileNamePdf = ZerofiltreUtils.sanitizeString(fullName) + "-" + ZerofiltreUtils.sanitizeString(courseTitle) + ".pdf";
+
+            if(certificatesStorageProvider.containsKey(fileNamePdf)) {
+                return certificatesStorageProvider.get(fileNamePdf);
+            }
+
+            String language = user.getLanguage() != null ? user.getLanguage() : Locale.FRANCE.getLanguage();
+
+            File certificate = createCertificate(new Locale(language), fullName, courseTitle, fileNamePdf);
+            certificatesStorageProvider.add(fileNamePdf, certificate);
+
+            return certificate;
+        }
+
+        throw new ZerofiltreException("Le certificat ne peut pas être délivré. Le cours n'est pas encore terminé.");
+    }
+
+    private File createCertificate(Locale locale, String fullName, String courseTitle, String fileNamePdf) throws IOException, ZerofiltreException {
+        Resource attachement = new ClassPathResource("mail_header_image.png");
+
+        Map<String, Object> templateModel = new HashMap<>();
+        templateModel.put("attachement", attachement.getURL().toString());
+        templateModel.put("fullName", fullName);
+        templateModel.put("courseTitle", courseTitle);
+        templateModel.put("today", LocalDate.now().format(DateTimeFormatter.ofPattern("dd/MM/yyyy")));
+        Context thymeleafContext = new Context();
+        thymeleafContext.setVariables(templateModel);
+        thymeleafContext.setLocale(locale);
+
+        String certificateHtml = templateEngine.process("certificate_course_completed.html", thymeleafContext);
+
+        log.debug("certificateHtml = {}", certificateHtml);
+
+        if(certificateHtml == null) throw new ZerofiltreException("Erreur lors de la génération du certificat.");
+
+        return convertHtmlToPdf(certificateHtml, fileNamePdf);
+    }
+
+    private File convertHtmlToPdf(String certificateHtml, String fileNamePdf) throws ZerofiltreException, IOException {
+        org.jsoup.nodes.Document documentJsoup = Jsoup.parse(certificateHtml);
+        documentJsoup.outputSettings().syntax(org.jsoup.nodes.Document.OutputSettings.Syntax.xml);
+        OutputStream outputStream;
+
+        try {
+            outputStream = new FileOutputStream(fileNamePdf);
+            ITextRenderer renderer = new ITextRenderer();
+            SharedContext sharedContext = renderer.getSharedContext();
+            sharedContext.setPrint(true);
+            sharedContext.setInteractive(false);
+            renderer.setDocumentFromString(documentJsoup.html());
+            renderer.layout();
+            renderer.createPDF(outputStream);
+            outputStream.close();
+        } catch (IOException e) {
+            throw new ZerofiltreException("Erreur lors de la génération du certificat.", e);
+        }
+
+        log.info("Course completion certificate converted to pdf completed.");
```

Comment:
log.info("Course completion certificate for file: {} converted to pdf completed.", fileName);